experimental/devices/as7262/as7262.go

```diff
@@ -221,17 +233,14 @@ func (d *Dev) Sense(ledDrive physic.ElectricCurrent, senseTime time.Duration) (S
 	}, nil
 }
 
-var waitForSensor = time.After
-
 // Halt stops any pending operations. Repeated calls to Halt do nothing.
 func (d *Dev) Halt() error {
-	d.canceledMu.Lock()
-	defer d.canceledMu.Unlock()
-
-	if !d.canceled {
-		d.done <- struct{}{}
-		d.canceled = true
-	}
+	d.cancelMu.Lock()
+	defer d.cancelMu.Unlock()
```

Comment:
There's a small race condition, you need to do `done := d.done` while holding d.cancelMu, so line 243 become `_, _ = <-done`, otherwise there's a tiny window where a concurrent Sense() call may replace d.done with a new channel. Not a big deal but I prefer this for complete correctness.

Reply:
I have the lock at both ends don't I? 
L136 and L238. Only other spot is in New()

Reply:
Ah I misread, I guess my point is that you don't need to hold cancelMu, but in practice it doesn't matter.